internify/backend/routes/api/jobs.js

```diff
@@ -1,55 +1,89 @@
-var express = require('express');
+var express = require("express");
 var router = express.Router();
-var JobPostingData = require('../../models/JobPostingData');
+var JobPostingData = require("../../models/JobPostingData");
 
 //@route    GET api/jobs/getAll
 //@desc     Get All Job Documents
 //@access   Public
-router.get('/', function(req, res, next) {
+router.get("/", function (req, res, next) {
   JobPostingData.find()
-    .then(jobs => res.status(200).json(jobs))
-    .catch(err => res.status(404).json({success: false}));
+    .then((jobs) => res.status(200).json(jobs))
+    .catch((err) => res.status(404).json({ success: false }));
+});
+
+//@route    GET api/jobs/getBulk
+//@desc     Get Bulk Job Documents
+//@access   Public
+router.get("/bulk/", function (req, res, next) {
+  const jobIds = req.query.data;
+  var promises = [];
+  var jobPostings = [];
+
+  function getJobPosting(jobId) {
+    return new Promise((resolve, reject) => {
+      JobPostingData.findOne({ jobId: jobId })
+        .then((job) => resolve(job))
+        .catch((err) => reject(err));
+    });
+  }
+
+  jobIds.forEach((jobId) => {
+    promises.push(getJobPosting(jobId));
+  });
+
+  Promise.all(promises)
+    .then((values) => {
+      jobPostings.push(values);
+    })
+    .catch((err) => {
+      res.status(404).json({ success: false });
+    })
+    .finally(() => {
+      res.status(200).json(jobPostings);
+    });
```

Comment:
The freaking 310 trauma that erected when I wrote this.

Reply:
beautiful though <3